core/commands/bootstrap.go

```diff
@@ -63,78 +62,54 @@ in the bootstrap list).
 		"default": bootstrapAddDefaultCmd,
 	},
 
-	Run: func(req cmds.Request, res cmds.Response) {
-		deflt, _, err := req.Option(defaultOptionName).Bool()
-		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
-		}
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		deflt, _ := req.Options[defaultOptionName].(bool)
 
 		var inputPeers []config.BootstrapPeer
 		if deflt {
 			// parse separately for meaningful, correct error.
 			defltPeers, err := config.DefaultBootstrapPeers()
 			if err != nil {
-				res.SetError(err, cmdkit.ErrNormal)
-				return
+				return err
 			}
 
 			inputPeers = defltPeers
 		} else {
-			parsedPeers, err := config.ParseBootstrapPeers(req.Arguments())
+			parsedPeers, err := config.ParseBootstrapPeers(req.Arguments)
 			if err != nil {
-				res.SetError(err, cmdkit.ErrNormal)
-				return
+				return err
 			}
 
 			inputPeers = parsedPeers
 		}
 
 		if len(inputPeers) == 0 {
-			res.SetError(errors.New("no bootstrap peers to add"), cmdkit.ErrClient)
-			return
+			return errors.New("no bootstrap peers to add")
 		}
 
-		r, err := fsrepo.Open(req.InvocContext().ConfigRoot)
+		ctx := env.(*oldcmds.Context)
```

Comment:
Not a requirement, but we should likely create another helper function in `core/commands/cmdenv` and call it `GetConfigRoot`.  This way it makes it clear that the cast is the right thing to do.  It would of saved me 10-15 minutes of review time.